django_apscheduler/jobstores.py

```diff
@@ -208,6 +208,21 @@ def lookup_job(self, job_id: str) -> Union[None, AppSchedulerJob]:
             return None
 
     def get_due_jobs(self, now) -> List[AppSchedulerJob]:
+        # Close old and unusable connections. Specifically:
+        # - Persistent connections past CONN_MAX_AGE.
+        # - Connections that errored out (e.g. due to the DB going down).
+        # If closed, the query in _get_jobs() will try to reconnect.
+        #
+        # Normally, Django calls this when a request starts and finishes,
+        # however we are (probably) not running in request context but in
+        # a continuous loop. So without this, the same connection would be
+        # used forever, and any error would never be recovered from (e.g.
+        # when the DB comes back up).
+        #
+        # get_due_jobs() seems like the most appropriate place for this,
+        # as it's called at the beginning of every scheduler iteration.
+        db.close_old_connections()
```

Comment:
Pre-emptively closing connections like this is problematic (see https://github.com/jcass77/django-apscheduler/issues/145#issuecomment-860250103).

The standard Django can get away with it because it does so at the beginning and end of a request-response-cycle, while all other db operations within that cycle still get to re-use the same connection.